spec/models/power_relay_spec.rb

```diff
@@ -11,11 +11,26 @@ class SomeRelay < Relay
 RSpec.describe SomeRelay do
 
   it { is_expected.to validate_presence_of :ip }
-  it { is_expected.to validate_presence_of :port }
+  it { is_expected.to validate_presence_of :outlet }
   it { is_expected.to validate_presence_of :username }
   it { is_expected.to validate_presence_of :password }
   it { is_expected.not_to validate_presence_of :auto_logout_minutes }
 
+  it "should allow a range of 1-16 outlets" do
+    relay = SomeRelay.new(ip: "123", username: "nucore", password: "password")
+
+    expect(relay.update(outlet: 16)).to be true
```

Comment:
This gets the job done, but I don't feel like it's very descriptive of what it's checking. Also, the `false` case could easily lead to a false positive where the object is invalid, but for the wrong reason.

It's a little more verbose, but I would consider one of these options. And since you're not hitting the database in the setup, I would lean towards one expectation per example.

```ruby
relay.outlet = 16
expect(relay).to be_valid

relay.outlet = 17
expect(relay).to be_invalid
# A couple options
expect(relay.errors).to include(:outlet)
expect(relay.errors[:outlet]).to include("is not included in the list") # I don't love this one because it relies on matching I18n
expect(relay.errors[:outlet]).to include(/included/)

# I like the clarity of this one, but can get annoying in that you need to include any interpolation arguments if the error includes them
expect(relay.errors).to be_added(:outlet, :inclusion)

Reply:
sure, I might argue that this route it becomes slightly more tedious should the maximum ever change in the future. but I have to admit the amount of clarity is rather nice and not something I'm used to seeing.